### ISSUE TITLE
RNAfusion: add clinical delivery upload

### ISSUE THREAD

**Issue by fevac** [Clinical-Genomics/cg#1961]
## Description

- Add clinical delivery upload for automatic delivery of fastq and analysis files

## Suggested solution

*Write here if there is a suggested way to solve the issue*

## This can be closed when

*Describe what needs to be done for this issue to be closed*

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

**Comment by seallard** [comment]
Is this related to the `@rnafusion.command("report-deliver")` @fevac? I noticed the `verify_analysis_finished` and `report_deliver` are not defined in that CLI command:

```python
@rnafusion.command("report-deliver")
@ARGUMENT_CASE_ID
@DRY_RUN
@click.pass_obj
def report_deliver(context: CGConfig, case_id: str, dry_run: bool) -> None:
    """Create a housekeeper deliverables file for given CASE ID."""

    analysis_api: AnalysisAPI = context.meta_apis["analysis_api"]

    try:
        analysis_api.verify_case_id_in_statusdb(case_id=case_id)
        analysis_api.verify_analysis_finished(case_id=case_id)
        if not dry_run:
            analysis_api.report_deliver(case_id=case_id)
        else:
            LOG.info("Dry-run")
    except CgError as error:
        LOG.error(f"Could not create report file: {error}")
        raise click.Abort()
    except Exception as error:
        LOG.error(f"Could not create report file: {error}")
        raise click.Abort()
```

**Comment by fevac** [comment]
No, this is related to `@click.command("clinical-delivery")`.  `verify_analysis_finished` and `report_deliver` are defined in meta

**Comment by seallard** [comment]
> report_deliver

Ok! So the command should use the functions defined in `RnafusionAnalysisAPI` in cg/meta/workflow/rnafusion.py? It it is not being resolved correctly in my editor at least, so I'm not sure it is actually finding those functions. Maybe it is because `analysis_api: AnalysisAPI` does not hold those functions. I should create a separate issue
